### DISCUSSION TITLE
Multiple USB devices on the bus

### DISCUSSION THREAD

**Discussion by Boneco0** [jblance/mpp-solar#185] [Q&A] [answered]
Hi,
I'm testing connection between my Ubuntu server and REVO II inverter and it's working when proper USB port is specified.
But I have few devices connected via USB ports. So after each restart they get different port number.
I tried to use symlink (udev) but i got an error:

**$ ll /dev/revo2**
lrwxrwxrwx 1 root root 7 Nov 28 21:54 /dev/revo2 -> ttyUSB2

**$ sudo mpp-solar -p /dev/revo2 -n 'REVO'  -c QPIGS**
2021-11-28 22:06:04,097:ERROR:device:run_command@286: No communications port defined - unable to run command QPIGS
Parameter                       Value           Unit
error                           No communications port defined - unable to run command QPIGS

**$ sudo mpp-solar -p /dev/ttyUSB2 -n 'REVO'  -c QPIGS**
Command: QPIGS - General Status Parameters inquiry
------------------------------------------------------------
Parameter                       Value           Unit
ac_input_voltage                237.0           V
ac_input_frequency              49.9            Hz
ac_output_voltage               237.0           V
ac_output_frequency             49.9            Hz
ac_output_apparent_power        0               VA
ac_output_active_power          285             W
ac_output_load                  9               %
bus_voltage                     401             V
battery_voltage                 25.6            V
battery_charging_current        1               A
battery_capacity                80              %
inverter_heat_sink_temperature  26              °C
pv_input_current_for_battery    0.0             A
pv_input_voltage                0.0             V
battery_voltage_from_scc        25.6            V
battery_discharge_current       0               A
is_sbu_priority_version_added   0               bool
is_configuration_changed        0               bool
is_scc_firmware_updated         0               bool
is_load_on                      1               bool
is_battery_voltage_to_steady_while_charging     0               bool
is_charging_on                  0               bool
is_scc_charging_on              0               bool
is_ac_charging_on               0               bool
rsv1                            0               A
rsv2                            0               A
pv_input_power                  0               W
is_charging_to_float            0               bool
is_switched_on                  1               bool
is_reserved                     0               bool

Is there way to use symlinks or other solution to automate port recognition?

**Comment by jblance** [comment, marked as answer]
use the ```--porttype serial``` to specify the type of port

**Reply by Boneco0** [reply]
Thank you :)
